sdk/python/feast/client.py

```diff
@@ -977,197 +601,131 @@ def ingest(
+        # Check 2) Check if FeatureTable batch source field mappings can be found in provided source table
+        _check_field_mappings(
+            column_names,
+            name,
+            feature_table.batch_source.timestamp_column,
+            feature_table.batch_source.field_mapping,
+        )
+        # Partition dataset by date
+        date_partition_dest_path = _partition_by_date(
+            column_names, feature_table, dest_path,
+        )
+
+        batch_source_type = SourceType(feature_table.batch_source.type).name
 
         try:
-            # Kafka configs
-            brokers = feature_set.get_kafka_source_brokers()
-            topic = feature_set.get_kafka_source_topic()
-            producer = get_producer(brokers, row_count, disable_progress_bar)
-
-            # Loop optimization declarations
-            produce = producer.produce
-            flush = producer.flush
-            ingestion_id = _generate_ingestion_id(feature_set)
-
-            # Transform and push data to Kafka
-            if feature_set.source.source_type == "Kafka":
-                for chunk in get_feature_row_chunks(
-                    file=dest_path,
-                    row_groups=list(range(pq_file.num_row_groups)),
-                    fs=feature_set,
-                    ingestion_id=ingestion_id,
-                    max_workers=max_workers,
-                ):
-
-                    # Push FeatureRow one chunk at a time to kafka
-                    for serialized_row in chunk:
-                        produce(topic=topic, value=serialized_row)
-
-                    # Force a flush after each chunk
-                    flush(timeout=timeout)
-
-                    # Remove chunk from memory
-                    del chunk
-
-            else:
-                raise Exception(
-                    f"Could not determine source type for feature set "
-                    f'"{feature_set.name}" with source type '
-                    f'"{feature_set.source.source_type}"'
+            if batch_source_type == "BATCH_FILE":
+                from urllib.parse import urlparse
+
+                file_url = feature_table.batch_source.file_options.file_url[:-1]
+                uri = urlparse(file_url)
+                staging_client = get_staging_client(uri.scheme)
+
+                file_paths = list()
+                for (dirpath, dirnames, filenames) in os.walk(date_partition_dest_path):
+                    file_paths += [os.path.join(dirpath, file) for file in filenames]
+                for path in file_paths:
+                    file_name = path.split("/")[-1]
+                    partition_date = path.split("/")[-2].split("=")[-1]
+                    staging_client.upload_file(
+                        path,
+                        uri.hostname,
+                        str(uri.path).strip("/")
+                        + "/"
+                        + f"date={partition_date}/"
```

Comment:
parquet structure should be used instead

Reply:
Updated.